### ISSUE TITLE
Adaption of labels for autoscaler forces replacement of CAPI cluster

### ISSUE THREAD

**Issue by Mazorius** [terraform-provider-openstack/terraform-provider-openstack#1625]
Hi there,

### Terraform Version

```shell
Terraform v1.6.4
on darwin_arm64
```

### Affected Resource(s)

- openstack_containerinfra_cluster_v1

### Terraform Configuration Files

-

### Debug Output

-

### Panic Output

-

### Expected Behavior

The label should be updated and the autoscaler should pick that new value to scale more or less worker nodes.

### Actual Behavior

The plan mentioned to replace the whole cluster only due to autoscaler update.

```shell
~ labels              = { # forces replacement
    ~ "max_node_count"                = "2" -> "4"
      # (4 unchanged elements hidden)
  }
```

### Steps to Reproduce

1. On an autoscaled cluster change the min/max node count to any lower or higher value.
2. `terraform plan -out plan.cache`

### Important Factoids

internal private Openstack cloud.

### References

I think it is the same as already done with the template_id, see #1598

**Comment by kayrus** [comment]
@Mazorius thanks for the bug report. Are you willing to provide a PR fixing this issue?

**Comment by Mazorius** [comment]
Yes, but I do not think that I will have some time for that this year.

If someone will implement that instead it would be great 😊

**Comment by pawcykca** [comment]
Hi @Mazorius,

Magnum API doesn't allow to change cluster's labels after cluster creation which is why terraform plan shows to replace the whole cluster in order to change `max_node_count` label.

The only way to change/update cluster labels after the cluster is created is to change the current cluster template with new one that contains the changed/updated labels, which results in a call of the Magnum API upgrade method - this was implemented by PR #1598. This API method currently implements (by Magnum Conductor component) only change of few labels regarding Kubernetes version upgrade and all other labels (including `max_node_count`) are ignored.

If you want to freely change minimum/maximum node count of nodegroups for autoscaling I would recommend to use `node-group-auto-discovery` parameter instead of `nodes` in Cluster Autoscaler configuration (Deployment definition) like below:
```
...
command:
...
  --node-group-auto-discovery=magnum:role=worker
...
```
This allows you to update minimum/maximum node count for each additional nodegroup created by terraform (using `openstack_containerinfra_nodegroup_v1` resource) by changing its `min_node_count` or `max_node_count` parameters.

Unfortunately this won't work for `default-worker` nodegroup which is created by default with cluster - here you will be able to change this parameters (`min_node_count` or `max_node_count`) only via Openstack CLI.
If you use Openstack version `Wallaby` or higher then you could create `default-worker` nodegroup scaled to 0 nodes and operate in terraform using additional nodegroups.

**Comment by mnaser** [comment]
I think the issue for this is that you need to be using the `min_node_count` and `max_node_count` paramters instead of using the `labels` with the labels inside of it.
